analysers/analyser_osmosis_relation_cyclic.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+
+###########################################################################
+##                                                                       ##
+## Copyrights Osmose Project 2022                                        ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+from modules.OsmoseTranslation import T_
+from .Analyser_Osmosis import Analyser_Osmosis
+
+sql10 = """
+WITH RECURSIVE rcte AS (
+  -- initial 'seed' relation
+  SELECT DISTINCT ON (relation_id)
+    relation_id AS start_id,
+    member_id,
+    1 as nloops,
+    CONCAT('r', relation_id) AS path
+  FROM
+    relation_members
+  WHERE
+    member_type = 'R' AND
+    relation_locate(relation_id) IS NOT NULL -- can't display relations without way/node members
+
+  UNION ALL
+
+  -- loop children until no 'relation'-type members or relation back to start
+  SELECT
+    rcte.start_id,
+    child.member_id,
+    nloops + 1 AS nloops,
+    CONCAT(rcte.path, ' > r', child.relation_id) AS path
+  FROM
+    rcte
+    JOIN relation_members AS child ON
+      child.relation_id = rcte.member_id
+    WHERE
```

Comment:
```suggestion
      child.relation_id = rcte.member_id AND
```

Reply:
Just curiosity, it seems either produces the same result (unlike `LEFT JOIN`s), is there a benefit to either of them? I picked this one as it seemed easier to read to me, but will adjust it. Just curious

Reply:
ON and WHERE are very close, but not the same. Here it just for code readability. I personally prefer ON for condition on joined table, keep condition and table declaration close.

Reply:
Thanks!